### ISSUE TITLE
cp: can't create directory No such file or directory

### ISSUE THREAD

**Issue by liudonghua123** [Cecilapp/GitHub-Pages-deploy#7]
I wrote the steps like the following.
```
      - name: Deploy to GitHub Pages
        uses: Cecilapp/GitHub-Pages-deploy@master
        env:
          EMAIL: liudonghua123@gmail.com
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          BUILD_DIR: build/web/
```

Then I got the following errors. I read the code, find there is an errors in https://github.com/Cecilapp/GitHub-Pages-deploy/blob/master/entrypoint.sh#L25.

```
 Deploy to GitHub Pages1s
cp: can't create directory '/github/home/build/web/': No such file or directory
Run Cecilapp/GitHub-Pages-deploy@master
  env:
    JAVA_HOME: /opt/hostedtoolcache/jdk/8.0.242/x64
    JAVA_HOME_8.0.242_x64: /opt/hostedtoolcache/jdk/8.0.242/x64
    FLUTTER_HOME: /opt/hostedtoolcache/flutter/1.14.6-beta/x64
    EMAIL: liudonghua123@gmail.com
    GH_TOKEN: ***
    BUILD_DIR: build/web/
/usr/bin/docker run --name e87b5238c4991814554b21bff138f123b62c93_1ab10a --label e87b52 --workdir /github/workspace --rm -e JAVA_HOME -e JAVA_HOME_8.0.242_x64 -e FLUTTER_HOME -e EMAIL -e GH_TOKEN -e BUILD_DIR -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/wayfless_generator_flutter/wayfless_generator_flutter":"/github/workspace" e87b52:38c4991814554b21bff138f123b62c93
### Started deploy to liudonghua123/wayfless_generator_flutter/gh-pages
cp: can't create directory '/github/home/build/web/': No such file or directory
```

**Comment by ArnaudLigny** [comment]
Hello @liudonghua123, good catch!

I think the copy fail cause of the subdir `build/` in `build/web/`.

I will try to fix that with the `--parents` option (https://linuxcommando.blogspot.com/2007/11/use-of-parents-flag-in-mkdir-and-c.html):

```bash
cp -R --parents $BUILD_DIR $HOME/$BUILD_DIR
```

**Comment by liudonghua123** [comment]
> Hello @liudonghua123, good catch!
> 
> I think the copy fail cause of the subdir `build/` in `build/web/`.
> 
> I will try to fix that with the `--parents` option (https://linuxcommando.blogspot.com/2007/11/use-of-parents-flag-in-mkdir-and-c.html):
> 
> ```shell
> cp -R --parents $BUILD_DIR $HOME/$BUILD_DIR
> ```

I think `--parents` option will not create directory if necessary, If I copy something to a directory which not exists, I will got `cp: with --parents, the destination must be a directory` error. So `mkdir` should be a must before `cp`.

**Comment by ArnaudLigny** [comment]
Fixed by #8.
